pkg/agent/discoverycache.go

```diff
@@ -202,3 +208,82 @@ func (j *discoveryCache) updateCategoryCache() {
 		}
 	}
 }
+
+func (j *discoveryCache) updateAccessRequestCache() {
+	log.Trace("updating AccessRequest cache")
+
+	// Update cache with published resources
+	existingApps := make(map[string]bool)
+	query := map[string]string{}
+
+	if !j.lastAccessRequestTime.IsZero() && !j.refreshAll {
+		query[apic.QueryKey] = fmt.Sprintf("state.name==%s", string(apic.AccessRequestProvisioned))
+	}
+	accessRequestInstances, _ := GetCentralClient().GetAPIV1ResourceInstancesWithPageSize(query, agent.cfg.GetSubscriptionURL(), apiServerPageSize)
+	accessRequests, err := v1alpha1.AccessRequestFromInstanceArray(accessRequestInstances)
+	if err != nil {
+		log.Error(utilErrors.Wrap(ErrUnableToGetAPIV1Resources, err.Error()).FormatError("AccessRequests"))
+		return
+	}
+
+	for _, accessReq := range accessRequests {
+		// Update the lastAccessRequestTime based on the newest access request found
+		thisTime := time.Time(accessReq.Metadata.Audit.CreateTimestamp)
+		if j.lastAccessRequestTime.Before(thisTime) {
+			j.lastAccessRequestTime = thisTime
+		}
+
+		externalAppID := addAccessRequestToAPICache(*accessReq)
+		existingApps[externalAppID] = true
+	}
+
+	if j.refreshAll {
+		// Remove items that are not published as Resources
+		cacheKeys := agent.cacheManager.GetAccessRequestCache().GetKeys()
+		for _, key := range cacheKeys {
+			if _, ok := existingApps[key]; !ok {
+				agent.cacheManager.GetAccessRequestCache().Delete(key)
+			}
+		}
+	}
+}
+
+func addAccessRequestToAPICache(accessReq v1alpha1.AccessRequest) string {
+	var (
+		externalAppID   string
+		externalAppName string
+	)
+	appID, ok := accessReq.Spec.Data["dataplaneAppID"]
+	externalAppID = appID.(string)
```

Comment:
could appID comeback nil or empty string here?

Reply:
It could be nil, I will add a check for this. This code will be used once we start tracking usages for subscriptions in the TAs